src/Microsoft.Azure.SignalR.Common/Endpoints/ServiceEndpoint.cs

```diff
@@ -17,6 +17,10 @@ public class ServiceEndpoint
 
         public string Endpoint => AccessKey?.Endpoint;
 
+        public string ServerEndpoint => Port.HasValue ? $"{Endpoint}:{Port}" : Endpoint;
```

Comment:
can port or Endpoint change during runtime? If not, set it when ctor instead of calculating it on every get. Same principle applies to other properties.